lib/utils/fs.go

```diff
@@ -241,13 +242,53 @@ func FSTryReadLockTimeout(ctx context.Context, filePath string, timeout time.Dur
 	return fileLock.Unlock, nil
 }
 
+// RemoveAllSecure is similar to [os.RemoveAll] but leverages [RemoveSecure] to delete files.
+func RemoveAllSecure(path string) error {
+	// Match os.RemoveAll protections in not permitting removal of "." directories
+	if path == "." || (len(path) >= 2 && path[len(path)-1] == '.' && os.IsPathSeparator(path[len(path)-2])) {
+		return &os.PathError{Op: "RemoveAllSecure", Path: path, Err: syscall.EINVAL} // error type matches os.RemoveAll
+	}
+
+	if info, err := os.Lstat(path); err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return trace.ConvertSystemError(err)
+	} else if !info.IsDir() {
+		// file or symlink
```

Comment:
Is there a point in doing RemoveSecure on a symlink? Should we just delete it as normal?

Reply:
I agree there is no value in a symlink case.  I documented it to be complete in the comment, but did not think it was a case that likely needed special handling.  Do you think it's worth the added code to handle symlinks differently?

Reply:
Your call, but lets test with symlinks regardless.

Reply:
Added a test, it works as expected so I suggest we just keep the code minimal

Reply:
Good, thanks!

Reply:
CI actually found a bug when the symlink points to a missing path.  So I may have to do explicit handling, looking now